guardian/manage-frontend
# Live chat mvp1

## What does this change?
This PR adds the live chat functionality. Once placed on a help centre page the `StartLiveChatButton` will initialise a chat between a user and a CSR. Clicking on this button will bring up a chat window and a user ending the chat will remove the window. This is in contrast to the default chat button behaviour where a sticky button persists in the bottom right corner of the screen regardless of whether a chat has been started or ended by a user.

`WithLiveChatContainer` is created so a user is able to maintain a connection to a CSR within the help centre. This component lives outside of the help centre router to ensure the chat persists and is high enough in the DOM that it is not removed when a user navigates help centre pages. This container is also styled to override the default css from the chat API.

The API settings require urls for images, as this would require further webpack/typescript configuration within the manage-frontend codebase images used for the chat window have been base64 encoded instead. Placeholder images have been used until further updates from design.

## How to test
- On this branch add the `StartLiveChatButton` button to a page in the help centre. 
- Either access the embedded service deployment settings in the salesforce DEV1 test sandbox or update the `liveChatAPI.init` function parameters with the relevant sandbox 
- Clicking the `StartLiveChatButton` button on the front end will raise a new case in the salesforce sandbox. Once the case is accepted the user should be able to chat with a CSR

## How can we measure success?
A user and CSR are able to connect and chat using the embedded service

## Images
| Minimised Chat  | Pre-chat Screen | Waiting to chat  | Chat in progress |
| ------------- | ------------- | ------------- | ------------- |
| ![image](https://user-images.githubusercontent.com/44685872/121561540-7d92eb00-ca10-11eb-958a-7bbaa1193528.png) | ![image](https://user-images.githubusercontent.com/44685872/121561668-a1eec780-ca10-11eb-8e2e-f6ae5d9c1938.png) | ![image](https://user-images.githubusercontent.com/44685872/121561888-d5c9ed00-ca10-11eb-814f-3f2a7f33a697.png) | ![image](https://user-images.githubusercontent.com/44685872/121562043-fabe6000-ca10-11eb-80a9-310a1a1836e1.png) |
